### ISSUE TITLE
[BUG] `split_record` output empty list for empty input string

### ISSUE THREAD

**Issue by ttnghia** [rapidsai/cudf#16453]
For an empty input string, `split_record` does not output anything (empty list). This is inconsistent with the cases when the input does not contain the split delimiter then the output list has one element equal to the input. For example:
```
input = "abcxyz"
output = split_record(input, "@")
output = ["abcxyz"]
```

However:
```
input = ""
output = split_record(input, "@")
output = []
```
This inconsistent behavior causes mismatching between cudf and Java String.split API, which states that:
> If the expression does not match any part of the input then the resulting array has just one element, namely this string.

Ref: https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#split-java.lang.String-int-

**Comment by ttnghia** [comment]
CC @davidwendt.

**Comment by davidwendt** [comment]
The output looks correct to me. It is returning a list with a single empty string and not an empty list.
The input string is empty and so it returns an empty string in the output.

```
>>> import cudf
>>> import pandas as pd
>>> ps = pd.Series(['','abcdef'])
>>> ps.str.split('@')
0          []
1    [abcdef]
dtype: object
>>> gs = cudf.Series(['','abcdef'])
>>> gs.str.split('@')
0          []
1    [abcdef]
dtype: list
```
The `[]` contains an empty string and does not indicate an empty list.
```
>>> ps.str.split('@')[0]
['']
>>> gs.str.split('@')[0]
['']
```

You can see this unit test also shows that any empty string will return an empty string
https://github.com/rapidsai/cudf/blob/211dbe43aafe9cd16ed9b85a72e67f1fcab65b91/cpp/tests/strings/split_tests.cpp#L246-L260
Note that the last entry is an empty string and the expected value for that row is `LCW{""}` and not `LCW{}`

**Comment by ttnghia** [comment]
The output seems to be correct in that case, in which the input has other non-empty strings.
For input with all empty, the output is empty: https://github.com/rapidsai/cudf/blob/211dbe43aafe9cd16ed9b85a72e67f1fcab65b91/cpp/tests/strings/split_tests.cpp#L310

**Comment by GregoryKimball** [comment]
Thank you @ttnghia for investigating this. We certainly have a bug in `split` and `rsplit` with an input column of all empty strings. @davidwendt and I discussed this issue and we agree that fixing the root cause will be too involved to make it in time for 24.08. We looked into some edge-case hacks on the libcudf side and couldn't find a working solution that would cover `split`, `split_records`, `split_re`, `rsplit`, etc. We plan to address this edge case bug in 24.10. 

Is it possible for Spark-RAPIDS to detect the all-empty string column case and avoid calling libcudf?
